File: tier1/java/b1873e10/Report.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

public final class Report {
    public static String build(String name) {
        String target = name;
        try {
            return new String(Files.readAllBytes(
                Paths.get("/srv/reports/" + target)));
        } catch (IOException e) {
            return "";
        }
    }
}

File: tier1/csharp/05dbd9fa/Report.cs
using System.IO;

public static class Report
{
    private const string Base = "/srv/reports";

    private sealed class Request
    {
        public string Name { get; set; }
    }

    public static string Build(string name)
    {
        var request = new Request();
        request.Name = Path.GetFileName(name);
        return Read(request);
    }

    private static string Read(Request request)
    {
        var target = Path.GetFullPath(Path.Combine(Base, request.Name));
        if (!target.StartsWith(Base + Path.DirectorySeparatorChar))
        {
            return "";
        }
        return File.ReadAllText(target);
    }
}
